python/pyarrow/_dataset.pyx

```diff
@@ -2299,6 +2299,16 @@ cdef class Scanner(_Weakrefable):
         ----------
         dataset : Dataset
             Dataset to scan.
+        use_threads : bool, default True
```

Comment:
For this one, I would maybe also change the order of the signature instead, and make all keywords (except the first) keyword-only, using `, *,`

Reply:
I pushed a change to address this in https://github.com/apache/arrow/pull/13983/commits/d04b2283ecee55cd1be02d2092a7b06f54b94514 that:

- Re-orders the signatures and docstrings inside the Scanner class to be consistent and follow what I think is a sensible order
- Makes `Scanner.from_dataset`, `Scanner.from_fragment`, and `Scanner.from_batches` use the above pattern to make all args except the first keyword only

Waiting on CI.